service/src/main/java/org/databiosphere/workspacedataservice/shared/model/BearerToken.java

```diff
@@ -0,0 +1,25 @@
+package org.databiosphere.workspacedataservice.shared.model;
+
+/**
+ * Record class to represent an auth token; use this instead of String for more type safety and
+ * readability of argument lists. As of this writing, WDS includes both Nimbus and Apache Oltu as
+ * transitive dependencies, and both of those libraries include models to represent a token; but
+ * they also both have more complexity than we need, and we don't want to depend too much on
+ * transitive dependencies
+ *
+ * @param value the String representation of the token
+ */
+public record BearerToken(String value) {
+
+  /**
+   * Validates that a given String is a valid access token. For the sake of WDS's validation, all we
+   * care about is that the String is non-null; we do not call out to an OAuth provider to validate
+   * expiration or authenticity.
+   *
+   * @param input the value to validate
+   * @return true if the value is valid
+   */
+  public static boolean isValid(String input) {
+    return input != null;
+  }
```

Comment:
Now that you have the `BearerToken` class, I think you also have a place to formalize what it means for the actual token to be `null`, give that thing a name, and encapsulate that concern instead of having to do null-checks anywhere.

I don't know if I have the name right here...but maybe it's as simple as `Unauthenticated`? :shrug:

```suggestion
  public static BearerToken unauthenticated() {
     return new BearerToken(null);
  }
  
  public boolean isAuthenticated() {
     return value != null;
  }
```

Reply:
Semantically, I want to be pretty clear about what we're actually providing here. What the current implementation offers is: "WDS found an `Authentication: Bearer … ` header, and saved its contents for later use". What I am trying to avoid is: "WDS has an OAuth token that is currently valid for use in Terra".

The latter implies that the token isn't expired, decodes as a JWT, has the right scopes, was issued by the right OAuth issuer, etc. I'm trying to keep WDS out of this business, and instead delegate to Sam and Azure b2c for this work.

The former makes very few promises. This keeps WDS simple: if it finds some opaque value in the http header, it will then pass that same http header onwards when it makes outbound requests.

Does this stance 1) make sense, and 2) change anything about the model you're suggesting?

Perhaps it's just a terminology thing, I could be satisfied with simply changing `BearerToken.isAuthenticated()` to `BearerToken.isEmpty()`.

Reply:
Totally agree with your separation of concerns here, keeping WDS lean and mean wrt to the assertions it makes about this token.  I'm mainly motivated to drive `null`s out the interaction here, and completely incapable of coming up with a good name for what you describe.  If you're good with `isEmpty` I am too!